src/compiler/checker.ts

```diff
@@ -28672,6 +28672,25 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         });
     }
 
+    function getThisTypeOfObjectLiteralFromContextualType(
+        containingLiteral: ObjectLiteralExpression,
+        contextualType = getApparentTypeOfContextualType(containingLiteral, /*contextFlags*/ undefined),

```

Comment:
there are only two callers of this function. I would drop the initialiser and change the second caller back to its original code that calculates it and then passes it in.